src/types/simple.rs

```diff
@@ -287,12 +287,12 @@ impl<'a> TryFrom<&'a SimpleType> for &'a ClassicType {
 #[cfg_attr(feature = "pyo3", pyclass)]
 #[non_exhaustive]
 #[serde(transparent)]
-pub struct TypeRow {
+pub struct TypeRow<T: PrimType> {
```

Comment:
We don't strictly need `PrimType` here - `Clone+'static+std::fmt::Debug` would do, but seemed a lot messier. Only when I hit [serialization](https://github.com/CQCL-DEV/hugr/pull/256/files#r1258318916) did I start to wonder.

Reply:
PrimType *is* now just `Clone + 'static + std::fmt::Debug`, with `const CLASSIC: bool` staying in `serialize.rs` :-)